### PR TITLE
Fix meta filelds duplication issue for hudi table

### LINKED ISSUE #4722: schema error in hudi external table when create external table like
<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
1. create external table like to create a hudi external table:
```
mysql> create external table hudi_crt_like_1 like ex_hudi_tbl;
Query OK, 0 rows affected (0.05 sec)
```
2. show create table sql from created hudi external table:
```
mysql> show create table hudi_crt_like_1;
+-----------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table           | Create Table                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
+-----------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| hudi_crt_like_1 | CREATE EXTERNAL TABLE `hudi_crt_like_1` (
  `uuid` int(11) NULL COMMENT "",
  `col_boolean` boolean NULL COMMENT "",
  `col_int` int(11) NULL COMMENT "",
  `col_long` bigint(20) NULL COMMENT "",
  `col_float` float NULL COMMENT "",
  `col_double` double NULL COMMENT "",
  `col_decimal` decimal128(38, 18) NULL COMMENT "",
  `col_date` date NULL COMMENT "",
  `col_string` varchar(65533) NULL COMMENT "",
  `_hoodie_commit_time` varchar(1048576) NULL COMMENT "",
  `_hoodie_commit_seqno` varchar(1048576) NULL COMMENT "",
  `_hoodie_record_key` varchar(1048576) NULL COMMENT "",
  `_hoodie_partition_path` varchar(1048576) NULL COMMENT "",
  `_hoodie_file_name` varchar(1048576) NULL COMMENT "",
  `_hoodie_commit_time` varchar(1048576) NULL COMMENT "",
  `_hoodie_commit_seqno` varchar(1048576) NULL COMMENT "",
  `_hoodie_record_key` varchar(1048576) NULL COMMENT "",
  `_hoodie_partition_path` varchar(1048576) NULL COMMENT "",
  `_hoodie_file_name` varchar(1048576) NULL COMMENT ""
) ENGINE=HUDI
COMMENT "PARTITION BY ()"
PROPERTIES (
"database" = "hudi_db",
"table" = "hudi_parquet_gzip",
"resource" = "hudi_emr_tn"
); |
+-----------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```
some columns created duplicated

### Expected behavior (Required)
no duplicated columns

### Real behavior (Required)
some columns created duplicated

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+------------------------+
| current_version()      |
+------------------------+
| QA_TEST_MASTER 66b5604 |
+------------------------+
1 row in set (0.01 sec)
```

### PR DESCRIPTION
**Pull request by miomiocat**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：
Closes #4722 

## Problem Summary(Required) ：
Fix meta filelds duplication issue if user has specified meta fields while creating hudi table,

### CODE REVIEW

**Review by dirtysalt — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 10 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Catalog.java|0|10|00.00%|

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java [hunk 1]
```diff
@@ -4392,11 +4393,21 @@ private void createIcebergTable(Database db, CreateTableStmt stmt) throws DdlExc
     private void createHudiTable(Database db, CreateTableStmt stmt) throws DdlException {
         String tableName = stmt.getTableName();
         List<Column> columns = stmt.getColumns();
-        columns.add(new Column("_hoodie_commit_time", Type.STRING, true));
-        columns.add(new Column("_hoodie_commit_seqno", Type.STRING, true));
-        columns.add(new Column("_hoodie_record_key", Type.STRING, true));
-        columns.add(new Column("_hoodie_partition_path", Type.STRING, true));
-        columns.add(new Column("_hoodie_file_name", Type.STRING, true));
+        if (columns.stream().noneMatch(c -> c.getName().equals(HoodieRecord.COMMIT_TIME_METADATA_FIELD))) {
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:4396]: set = columns.stream().filter(c -> sets.contains(c.getName())).collect(Collectors.toSet());
sets.stream().foreach { 
  s -> if (!set.contains(s)) {
 columns.add(new Column(s, Type.STRING, true));
}
};

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java [hunk 2]
```diff
@@ -4392,11 +4393,21 @@ private void createIcebergTable(Database db, CreateTableStmt stmt) throws DdlExc
...
+            columns.add(new Column("_hoodie_commit_time", Type.STRING, true));
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:4397]: COMMIT_TIME_METADATA_FIELD

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java (hunk 1 above)
Comment (miomiocat) [reply fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:4408]: change to (all-set SUB include-set)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 5 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Catalog.java|0|5|00.00%|

**Review by caneGuy — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Comment by miomiocat** [comment]
@Mergifyio backport branch-2.2

**Pull request merged**